File: API/Entities/OrderAggregate/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace API.Entities.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; } 
        public string BuyerId { get; set; }
        [Required]
        public ShippingAddress shippingAddress{ get; set; }
        public DateTime OrderDate { get; set; }= DateTime.Now;
        public List<OrderItem> orderItems{ get; set; }
        public long SubTotal { get; set; }
        public long DeliveryFee { get; set; }
        public OrderStatus orderStatus{ get; set; } = OrderStatus.pending;

        public long GetTotal()
        {
            return SubTotal + DeliveryFee;
        }
    }                                                                                                                                                                                      
}

File: client/src/app/models/Order.ts
import { unstable_DataStrategyFunctionArgs } from "react-router-dom"

export interface Order {
    id: number
    buyerId: string
    shippingAddress: ShippingAddress
    orderDate: string
    orderItems: OrderItem[]
    subTotal: number
    deliveryFee: number
    orderStatus: number
    total: number
  }
  
  export interface ShippingAddress {
    fullName: string
    address1: string
    address2: string
    city: string
    state: string
    zipCode: string
    country: string
  }
  
  export interface OrderItem {
    productId: number
    name: string
    pictureUrl: string
    price: number
    quantity: number
  }
//   export enum orderStatus
//   {
//     Pending,
//     PaymentRecieved,
//     PaymentFailed
//   }
  